### DISCUSSION TITLE
Question about how Copilot clients work in general

### DISCUSSION THREAD

**Discussion by davidmichaelkarr** [Genuitec/Copilot4Eclipse-community#8] [Help] [answered]
Isn't it the case that the copilot4eclipse plugin is just implementing a protocol to talk to Copilot, and that Github doesn't need to have any knowledge of what client this is, just that it is fulfilling the required protocol?

To be clear, I'm not reporting a problem, just trying to get some information.

**Comment by wayneparrott** [comment, marked as answer]
Correct, Copilot4Eclipse interacts with GitHub Copilot services over secure network connection. The protocol is based on the Language Server Protocol (LSP). A Copilot session includes user identification and authentication as well as api client details.

W
